primefaces/src/main/java/org/primefaces/component/badge/BadgeRenderer.java

```diff
@@ -47,7 +47,7 @@ public static <T extends UIComponent> void encodeOverlayed(FacesContext context,
             IOBiConsumer<FacesContext, T> contentRenderer, T component) throws IOException {
         BadgeModel badgeModel = Badge.getBadgeModel(badge);
         if (badgeModel != null) {
-            BadgeRenderer badgeRenderer = Badge.getDefaultRenderer(context);
+            BadgeRenderer badgeRenderer = new Badge().getRenderer();
```

Comment:
i dont understand this in general
Badge#geteEnderer should just return BadgeRenderer, "this"?

Reply:
nevermind
that code it a bit weird with statics, non-statics, functional and so on

we need at least use #createComponent API instead of new Badge

later we may refactor that part a bit

Reply:
Yeah, I just came back to this because we also constructed a renderer there. The thing with badges is that you want to be able to render a badge when you only have a badge model object. For example with a SpeedDial. We can improve this later.